SSF wiki page (#243)
eth-protocol-fellows/protocol-studies: docs/wiki/research/SSF.md

# Single slot finality (SSF)

## Roadmap tracker

| Upgrade |   URGE    | Track |          Topic           |                                               Cross-references                                               |
| :-----: | :-------: | :---: | :----------------------: | :----------------------------------------------------------------------------------------------------------: |

> |   SSF   | the Merge |   -   | Increase Validator count | intersection with: [MAX_EB](/docs/wiki/research/cl-upgrades.md), [ePBS](/docs/wiki/research/PBS/ePBS.md), ET |

Author: Checking Vs picture again. It seems that SSF *is* part of the increasing validator count track. No topic tho, since there's only one topic and that overlaps with the track (in contrast to other urges e.g. scourge where there are multiple topics spread over 2 tracks).
So, I propose 
- track : increasing validator count
- topic : - 

Regarding intersections: SSLE seems legit. But I would keep max_eb (closely related to enabling greater validator count), and yes PBS

> |   SSF   | the Merge |   -   | Increase Validator count | intersection with: [MAX_EB](/docs/wiki/research/cl-upgrades.md), [ePBS](/docs/wiki/research/PBS/ePBS.md), ET |

Reviewer: The overall track here should be something like PoS Improvements and topic more specific, something like faster finalization/economic security. I don't want to leave out empty columns here, otherwise I would make them optional or remove in the template for these tables.

The picture from V shouldn't be an ultimate reference tool because it's changing and doesn't reflect all realities. 
SSF makes it harder to increase validator count, the more validators, the harder the aggregation in one slot. The idea in the picture is to build on top of this to make it more effective.

## What is SSF?

> [Single Slot Finality](/docs/eps/week10-research.md) is a concept within Ethereum, in the context of consensus mechanism, that addresses the inefficiencies associated with the time it takes to finalize blocks, proposing a significant raise in blocks validation efficiency and a drastic reduction of time-to-finality. 

Reviewer: ```suggestion
[Single Slot Finality](/docs/eps/week10-research.md) is a research concept of an improvement of Beacon Chain's consensus mechanism that addresses inefficiencies associated with the time it takes to finalize blocks. It's proposing a significant raise in blocks validation efficiency and a drastic reduction of time-to-finality. 
```

Instead of waiting for 2 epochs, blocks could get proposed and finalized in the same slot.

## Context and Motivation

> Ethereum finalizes every 2 epochs, or every 64 slots. With each slot being 12 seconds long, finalization takes around 12.8 minutes, at the moment of writing.

Reviewer: Before this sentence, there should be also an intro to what is finality/weak subjectivity. Something along:

```suggestion
Ethereum consensus layer implements Gasper protocol which includes Casper Friendly Finality Gadget. Casper FFG ensures that the network keeps producing blocks and accumulates validator attentions for each epoch. Finality is the ultimate state of PoS economic security and its change would require 2/3 of the validator set to be slashed. 
Beacon Chain achieves finality every 2 epochs, or every 64 slots. With each slot being 12 seconds long, finalization takes around 12.8 minutes, at the moment of writing.
```

Reviewer: Maybe also add a reference to PoS docs explaining finality

This current time to finality has turned out to be too long for most users, and is inconvenient for apps and exchanges that might not want to wait that long to be certain their transactions are permanent. 
The delay between a block's proposal and finalization also creates an opportunity for short reorgs that an attacker could use to censor certain blocks or extract MEV.

## Benefits of SSF

> * more convenient for apps - transactions finalization time improved by an order of magnitude, i.e. 12 seconds instead of 12 minutes means better UX for all Ethereum users

Reviewer: Use capital letters even in bullet points

The fork choice algorithm only considers blocks since the last finalized block. Under SSF there will not be any blocks for the fork choice rule to consider, because finality occurs in the same slot as the block is proposed. This means that under SSF **either** the fork choice algorithm **or** the finality gadget would be active at any time. 
The finality gadget will finalize blocks where $2/3$ of validators were online and attesting honestly. If a block is not able to exceed the $2/3$ threshold, the fork choice rule would kick in to determine which chain to follow. This also creates an opportunity to maintain the inactivity leak mechanism that recovers a chain where $>1/3$ validators go offline. If a block is not able to exceed the $2/3$ threshold, the fork choice rule would kick in to determine which chain to follow.

Some interaction issues between the fork choice and the consensus do remain in any such design, and it’s still important to work through them. 
Short-term improvements to the existing fork choice (eg. view-merge) may also feed into work on the SSF fork choice.[^2]

> 

Reviewer: Feel free to use drawings from Francesco in this part, he shared the raw drawio file so you can edit it
